pypeit/spectrographs/keck_esi.py

```diff
@@ -0,0 +1,404 @@
+        """
+        Methods to generate metadata requiring interpretation of the header
+        data, instead of simply reading the value of a header card.
+
+        Args:
+            headarr (:obj:`list`):
+                List of `astropy.io.fits.Header`_ objects.
+            meta_key (:obj:`str`):
+                Metadata keyword to construct.
+
+        Returns:
+            object: Metadata value read from the header(s).
+        """
+        if meta_key == 'binning':
+            binspatial, binspec = parse.parse_binning(headarr[0]['BINNING'])
+            return parse.binning2string(binspec, binspatial)
+        elif meta_key == 'dispname':
+            dname = headarr[0]['INSTRUME'].split(' ')[1]
+            return dname
+        else:
+            msgs.error("Not ready for this compound meta")
+
+    def configuration_keys(self):
+        """
+        Return the metadata keys that define a unique instrument
+        configuration.
+
+        This list is used by :class:`~pypeit.metadata.PypeItMetaData` to
+        identify the unique configurations among the list of frames read
+        for a given reduction.
+
+        Returns:
+            :obj:`list`: List of keywords of data pulled from file headers
+            and used to constuct the :class:`~pypeit.metadata.PypeItMetaData`
+            object.
+        """
+        return []
+
+    def check_frame_type(self, ftype, fitstbl, exprng=None):
+        """
+        Check for frames of the provided type.
+
+        Args:
+            ftype (:obj:`str`):
+                Type of frame to check. Must be a valid frame type; see
+                frame-type :ref:`frame_type_defs`.
+            fitstbl (`astropy.table.Table`_):
+                The table with the metadata for one or more frames to check.
+            exprng (:obj:`list`, optional):
+                Range in the allowed exposure time for a frame of type
+                ``ftype``. See
+                :func:`pypeit.core.framematch.check_frame_exptime`.
+
+        Returns:
+            `numpy.ndarray`_: Boolean array with the flags selecting the
+            exposures in ``fitstbl`` that are ``ftype`` type frames.
+        """
+        if ftype in ['pinhole', 'dark']:
+            # No pinhole or pinhole or dark frames
+            return np.zeros(len(fitstbl), dtype=bool)
+        if ftype in ['bias']:
+            return fitstbl['idname'] == 'Bias'
+        if ftype in ['pixelflat', 'trace', 'illumflat']:
+            ans = np.zeros(len(fitstbl), dtype=bool)
+            for kk, idnm in enumerate(fitstbl['idname']):
+                if idnm in ['DmFlat', 'IntFlat', 'SkyFlat']:
+                    ans[kk] = True
+            return ans
+        if ftype in ['arc', 'tilt']:
+            return fitstbl['idname'] == 'Line'
+        if ftype == 'science':
+            return (fitstbl['idname'] == 'Object') \
+                        & framematch.check_frame_exptime(fitstbl['exptime'], (60., None)) 
```

Comment:
instead of hard-coding the exposure time here, should you just add 
`good_exp = framematch.check_frame_exptime(fitstbl['exptime'], exprng)` at the beginning of this method and then add the `good_exp` flag here (as we do with other instruments)? You already have set up the default exposure times in `default_pypeit_par()`, but they are different than these one.

Reply:
done